rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2209,15 +2208,41 @@ void getProposedFederation_whenStorageProviderReturnsProposedFederation_shouldRe
         assertEquals(proposedFederation, actualProposedFederation.get());
     }
 
+    @Test
+    void getProposedFederatorPublicKeyOfType_whenFederationIsEmpty_shouldReturnEmpty() {
+        // Act
+        Optional<byte[]> actualPublicKey = federationSupport.getProposedFederatorPublicKeyOfType(0, FederationMember.KeyType.BTC);
+
+        // Assert
+        assertFalse(actualPublicKey.isPresent());
+    }
+
+    @Test
+    void getProposedFederatorPublicKeyOfType_whenStorageProviderReturnsProposedFederation_shouldReturnPublicKey() {
+        // Arrange
+        Federation proposedFederation = P2shErpFederationBuilder.builder().build();
+        storageProvider.setProposedFederation(proposedFederation);
+
+        FederationMember.KeyType keyType = FederationMember.KeyType.BTC;
```

Comment:
lets make tests for RSK and MST keys too

Reply:
Sounds good, will update

Reply:
Updated